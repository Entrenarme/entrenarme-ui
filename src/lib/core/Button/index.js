// @flow
import * as React from 'react';
import MButton from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import styled from 'styled-components';

import * as fonts from '../../helpers/fonts';
import colors from '../../helpers/colors';

type ButtonProps = {
  options: {
    type: ButtonType,
    state: ButtonState,
    colorVariant: ButtonColorVariant,
    disabled: ButtonDisabled,
  },
};

const getBgColor = (
  buttonType: ButtonType,
  options: 'default' | 'active',
  state: ButtonState,
  disabled: ButtonDisabled,
  colorVariant: ButtonColorVariant,
) => {
  if (buttonType === 'link' || buttonType === 'outline') {
    return colors.white.default;
  }
  if (buttonType === 'add' || disabled) {
    return colors.gray.default;
  }
  if (state === 'error' || state === 'success') {
    return colors[state].default;
  }
  return colors[colorVariant || state].default;
};

const getColor = (
  buttonType: ButtonType,
  state: ButtonState,
  colorVariant: ButtonColorVariant,
  disabled: ButtonDisabled,
) => {
  if (buttonType === 'outline') {
    return colors[colorVariant].default;
  }
  if (disabled) {
    return colors.gray.dark;
  }
  if (state === 'loading') {
    if (buttonType === 'link') {
      return colors.gray.loading;
    }
    if (buttonType === 'add') {
      return colors.secondary.loading;
    }
    return colors.white.loading;
  }
  if (buttonType === 'link' || buttonType === 'add') {
    return colors.secondary.default;
  }
  return colors.white.default;
};

const SButton = styled(MButton)`
  background-color: ${(props: ButtonProps) =>
    getBgColor(
      props.options.type,
      'default',
      props.options.state,
      props.options.disabled,
      props.options.colorVariant,
    )} !important;
  ${(props: ButtonProps) =>
    props.options.type === 'outline'
      ? `	border: 2px solid ${getColor(
          props.options.type,
          props.options.state,
          props.options.colorVariant,
          props.options.disabled,
        )} !important`
      : props.options.type === 'add'
        ? `border: 2px dashed ${colors.gray.dark} !important;`
        : ''};
  :hover {
    background-color: ${(props: ButtonProps) =>
      getBgColor(
        props.options.type,
        'active',
        props.options.state,
        props.options.disabled,
        props.options.colorVariant,
      )} !important;
  }
  span {
    color: ${(props: ButtonProps) =>
      getColor(
        props.options.type,
        props.options.state,
        props.options.colorVariant,
        props.options.disabled,
      )};
    font-family: ${fonts.mainFont};
  }
`;

type Props = {
  /** This is what is gonig to be rendered inside the button */
  children: React.Node,
  /** onClick handler passed down to the button */
  onClick?: Function,
  /** type of the button, 'default' | 'link' | 'add' | 'outline' */
  type?: ButtonType,
  /** state of the button, 'outline' | 'disabled' | 'loading' | 'icon' */
  state?: ButtonState,
  /** the color of the button, | 'primary' | 'secondary' | 'warning' | 'error' | 'success' */
  colorVariant?: ButtonColorVariant,
  /** the element to be placed on the button */
  icon?: React.Node,
  /** if the button is in disabled state */
  disabled?: ButtonDisabled,
  /** text to be rendered on the loading state */
  loadingText?: React.Node,
  /** text to be rendered on the error state */
  errorText?: React.Node,
  /** text to be rendered on the success state */
  successText?: React.Node,
};

const Icon = styled.span`
  margin-right: 10px;
  height: 18px;
`;

const textToRender = (
  state?: ButtonState,
  children: React.Node,
  loadingText: React.Node,
  errorText: React.Node,
  successText: React.Node,
) => {
  switch (state) {
    case 'loading':
      return loadingText;
    case 'error':
      return errorText;
    case 'success':
      return successText;
    default:
      return children;
  }
};

const Button = ({
  children,
  type,
  state,
  icon,
  colorVariant,
  disabled,
  defaultText,
  loadingText,
  errorText,
  successText,
  ...rest
}: Props) => (
  <SButton
    options={{ type, state, colorVariant, disabled }}
    disabled={
      state === 'loading' ||
      state === 'error' ||
      state === 'success' ||
      disabled
    }
    {...rest}
  >
    {icon ? <Icon>{icon}</Icon> : null}{' '}
    {state === 'loading' ? (
      <CircularProgress
        style={{
          height: '18px',
          width: '18px',
          marginRight: '10px',
          color:
            type === 'add'
              ? colors.secondary.loading
              : type === 'outline'
                ? colors[colorVariant].default
                : colors.gray.loading,
        }}
      />
    ) : null}{' '}
    <span>
      {textToRender(state, children, loadingText, errorText, successText)}
    </span>
  </SButton>
);

Button.defaultProps = {
  onClick: () => {},
  type: 'default',
  state: 'default',
  icon: null,
  disabled: false,
  colorVariant: 'primary',
  loadingText: '',
  errorText: '',
  successText: '',
};

export default Button;
