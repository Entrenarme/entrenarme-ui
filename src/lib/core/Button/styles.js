// @flow
import styled from 'styled-components';
import MButton from '@material-ui/core/Button';

import * as fonts from '../../helpers/fonts';
import colors from '../../helpers/colors';

import { getBgColor, getColor } from './utils';

type ButtonProps = {
  options: {
    type: ButtonType,
    state: ButtonState,
    colorVariant: ButtonColorVariant,
    disabled: ButtonDisabled,
  },
};

export const SButton = styled(MButton)`
  height: 40px;
  box-sizing: border-box;
  background-color: ${(props: ButtonProps) =>
    getBgColor(
      props.options.type,
      'default',
      props.options.state,
      props.options.disabled,
      props.options.colorVariant,
    )} !important;
  ${(props: ButtonProps) =>
    props.options.type === 'outline'
      ? `	border: 2px solid ${getColor(
          props.options.type,
          props.options.state,
          props.options.colorVariant,
          props.options.disabled,
        )} !important`
      : props.options.type === 'add'
        ? `border: 2px dashed ${colors.gray.dark} !important;`
        : ''};
  :hover {
    background-color: ${(props: ButtonProps) =>
      getBgColor(
        props.options.type,
        'active',
        props.options.state,
        props.options.disabled,
        props.options.colorVariant,
      )} !important;
  }
  span {
    color: ${(props: ButtonProps) =>
      getColor(
        props.options.type,
        props.options.state,
        props.options.colorVariant,
        props.options.disabled,
      )};
    font-family: ${fonts.mainFont};
  }
`;

export const Icon = styled.span`
  margin-right: 10px;
  height: 18px;
`;
