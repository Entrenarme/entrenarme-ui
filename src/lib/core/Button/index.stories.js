// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

import Button from './index';

const actions = {
  onClick: action('onClick'),
};

const createButton = (
  {
    type = 'primary',
    state,
    colorVariant,
    disabled,
    icon,
  }: {
    type?: ButtonType,
    state?: ButtonState,
    colorVariant?: ButtonColorVariant,
    disabled?: ButtonDisabled,
    icon?: React.Node,
  } = { colorVariant: 'primary' },
) => (
  <Button
    type={type}
    state={state}
    colorVariant={colorVariant}
    disabled={disabled}
    icon={icon}
    loadingText="loading"
    errorText="error"
    successText="success"
    {...actions}
  >
    button {colorVariant}
  </Button>
);

storiesOf('Buttons/Default', module)
  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
  .add('default (primary)', withInfo('Button info')(() => createButton()))
  .add(
    'secondary',
    withInfo('Button secondary')(() =>
      createButton({ colorVariant: 'secondary' }),
    ),
  )
  .add(
    'warning',
    withInfo('Button warning')(() => createButton({ colorVariant: 'warning' })),
  )
  .add(
    'loading',
    withInfo('Button loading')(() => createButton({ state: 'loading' })),
  )
  .add(
    'error color',
    withInfo('Button error')(() => createButton({ colorVariant: 'error' })),
  )
  .add(
    'error state',
    withInfo('Button error')(() => createButton({ state: 'error' })),
  )
  .add(
    'success color',
    withInfo('Button success')(() => createButton({ colorVariant: 'success' })),
  )
  .add(
    'success state',
    withInfo('Button success')(() => createButton({ state: 'success' })),
  )
  .add(
    'icon',
    withInfo('Button icon')(() =>
      createButton({
        icon: <FontAwesomeIcon icon={faCoffee} />,
      }),
    ),
  );

storiesOf('Buttons/Outline', module)
  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
  .add(
    'default (primary)',
    withInfo('Button info')(() =>
      createButton({ colorVariant: 'primary', type: 'outline' }),
    ),
  )
  .add(
    'secondary',
    withInfo('Button outline secondary')(() =>
      createButton({ colorVariant: 'secondary', type: 'outline' }),
    ),
  )
  .add(
    'warning',
    withInfo('Button outline warning')(() =>
      createButton({ colorVariant: 'warning', type: 'outline' }),
    ),
  )
  .add(
    'loading',
    withInfo('Button outline loading')(() =>
      createButton({ type: 'outline', state: 'loading' }),
    ),
  )
  .add(
    'error',
    withInfo('Button outline error')(() =>
      createButton({ type: 'outline', colorVariant: 'error' }),
    ),
  )
  .add(
    'success',
    withInfo('Button outline success')(() =>
      createButton({ type: 'outline', colorVariant: 'success' }),
    ),
  )
  .add(
    'icon',
    withInfo('Button outline icon')(() =>
      createButton({
        type: 'outline',
        icon: <FontAwesomeIcon icon={faCoffee} />,
      }),
    ),
  );

storiesOf('Buttons/Link', module)
  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
  .add(
    'default',
    withInfo('Button link default')(() => createButton({ type: 'link' })),
  )
  .add(
    'disabled',
    withInfo('Button link disabled')(() =>
      createButton({ type: 'link', disabled: true }),
    ),
  )
  .add(
    'loading',
    withInfo('Button link loading')(() =>
      createButton({ type: 'link', state: 'loading' }),
    ),
  )
  .add(
    'icon',
    withInfo('Button link icon')(() =>
      createButton({
        type: 'link',
        icon: <FontAwesomeIcon icon={faCoffee} />,
      }),
    ),
  );

storiesOf('Buttons/Add', module)
  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
  .add(
    'default',
    withInfo('Button add default')(() => createButton({ type: 'add' })),
  )
  .add(
    'disabled',
    withInfo('Button add disabled')(() =>
      createButton({ type: 'add', disabled: true }),
    ),
  )
  .add(
    'loading',
    withInfo('Button add loading')(() =>
      createButton({ state: 'loading', type: 'add' }),
    ),
  )
  .add(
    'icon',
    withInfo('Button add icon')(() =>
      createButton({
        type: 'add',
        icon: <FontAwesomeIcon icon={faCoffee} />,
      }),
    ),
  );
