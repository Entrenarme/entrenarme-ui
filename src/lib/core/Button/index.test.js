import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCoffee } from '@fortawesome/pro-light-svg-icons';

import Button from './index';

test('renders default with no props', () => {
  const { container, rerender } = render(<Button>default</Button>);
  expect(container.firstChild).toMatchSnapshot('default button');

  rerender(<Button colorVariant="secondary">secondary</Button>);
  expect(container.firstChild).toMatchSnapshot('default secondary button');

  rerender(<Button colorVariant="warning" successText="success" />);
  expect(container.firstChild).toMatchSnapshot('default warning button');

  rerender(<Button state="loading" loadingText="loading" />);
  expect(container.firstChild).toMatchSnapshot('default loading button');

  rerender(
    <Button colorVariant="error" errorText="Error text">
      error color
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('default error color button');

  rerender(<Button state="error" errorText="error" />);
  expect(container.firstChild).toMatchSnapshot('default error button');

  rerender(
    <Button colorVariant="success" errorText="Success text">
      success color
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('default success color button');

  rerender(<Button state="success" successText="success" />);
  expect(container.firstChild).toMatchSnapshot('default success button');

  rerender(<Button disabled>default disabled</Button>);
  expect(container.firstChild).toMatchSnapshot('default disabled button');

  rerender(
    <Button icon={<FontAwesomeIcon icon={faCoffee} />}>
      default with icon
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('default icon button');
});

test('renders outline', () => {
  const { container, rerender } = render(
    <Button type="outline">outline</Button>,
  );
  expect(container.firstChild).toMatchSnapshot('outline default');

  rerender(
    <Button type="outline" colorVariant="secondary">
      secondary
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('outline secondary button');

  rerender(
    <Button type="outline" colorVariant="warning" successText="success" />,
  );
  expect(container.firstChild).toMatchSnapshot('outline warning button');

  rerender(<Button type="outline" state="loading" loadingText="loading" />);
  expect(container.firstChild).toMatchSnapshot('outline loading button');

  rerender(
    <Button type="outline" colorVariant="error" errorText="Error text">
      error color
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('outline error color button');

  rerender(<Button type="outline" state="error" errorText="error" />);
  expect(container.firstChild).toMatchSnapshot('outline error button');

  rerender(
    <Button type="outline" colorVariant="success" errorText="Success text">
      success color
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('outline success color button');

  rerender(<Button type="outline" state="success" successText="success" />);
  expect(container.firstChild).toMatchSnapshot('outline success button');

  rerender(
    <Button type="outline" disabled>
      default disabled
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('outline disabled button');

  rerender(
    <Button type="outline" icon={<FontAwesomeIcon icon={faCoffee} />}>
      default with icon
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('outline icon button');
});

test('renders link', () => {
  const { container, rerender } = render(<Button type="link">link</Button>);
  expect(container.firstChild).toMatchSnapshot('link default');

  rerender(<Button type="link" state="loading" loadingText="loading" />);
  expect(container.firstChild).toMatchSnapshot('link loading button');

  rerender(
    <Button type="link" disabled>
      default disabled
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('link disabled button');

  rerender(
    <Button type="link" icon={<FontAwesomeIcon icon={faCoffee} />}>
      default with icon
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('link icon button');
});

test('renders add', () => {
  const { container, rerender } = render(<Button type="add">add</Button>);
  expect(container.firstChild).toMatchSnapshot();

  rerender(<Button type="add" state="loading" loadingText="loading" />);
  expect(container.firstChild).toMatchSnapshot('add loading button');

  rerender(
    <Button type="add" disabled>
      default disabled
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('add disabled button');

  rerender(
    <Button type="add" icon={<FontAwesomeIcon icon={faCoffee} />}>
      default with icon
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot('add icon button');
});
