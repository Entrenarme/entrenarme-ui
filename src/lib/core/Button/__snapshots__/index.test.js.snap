// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add disabled button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-203 MuiButtonBase-disabled-204 MuiButton-root-179 MuiButton-text-181 MuiButton-flat-184 MuiButton-disabled-197 sc-bdVaJa cNrUNG"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-180"
  >
    <span>
      default disabled
    </span>
  </span>
</button>
`;

exports[`add icon button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-203 MuiButton-root-179 MuiButton-text-181 MuiButton-flat-184 sc-bdVaJa cbcxry"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-180"
  >
    <span
      class="sc-bwzfXH czactv"
    >
      <svg
        aria-hidden="true"
        data-prefix="fal"
        data-icon="coffee"
        class="svg-inline--fa fa-coffee fa-w-18 "
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 576 512"
      >
        <path
          fill="currentColor"
          d="M517.9 448H26.1c-24.5 0-33.1-32-20-32h531.8c13.1 0 4.5 32-20 32zM576 159.1c.5 53.4-42.7 96.9-96 96.9h-32v32c0 53-43 96-96 96H160c-53 0-96-43-96-96V76c0-6.6 5.4-12 12-12h402.8c52.8 0 96.7 42.2 97.2 95.1zM416 96H96v192c0 35.3 28.7 64 64 64h192c35.3 0 64-28.7 64-64V96zm128 64c0-35.3-28.7-64-64-64h-32v128h32c35.3 0 64-28.7 64-64z"
        />
      </svg>
    </span>
    <span>
      default with icon
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-222"
  />
</button>
`;

exports[`add loading button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-203 MuiButtonBase-disabled-204 MuiButton-root-179 MuiButton-text-181 MuiButton-flat-184 MuiButton-disabled-197 sc-bdVaJa kxuXFA"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-180"
  >
    <div
      class="MuiCircularProgress-root-213 MuiCircularProgress-colorPrimary-216 MuiCircularProgress-indeterminate-215"
      style="margin-right: 10px; color: rgba(0, 181, 226, 0.6);"
      role="progressbar"
    >
      <svg
        class="MuiCircularProgress-svg-218"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle-219 MuiCircularProgress-circleIndeterminate-221"
          cx="44"
          cy="44"
          r="20.2"
          fill="none"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <span>
      loading
    </span>
  </span>
</button>
`;

exports[`default button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bdVaJa gjZMZu"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-2"
  >
    <span>
      default
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-28"
  />
</button>
`;

exports[`default disabled button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-25 MuiButtonBase-disabled-26 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 MuiButton-disabled-19 sc-bdVaJa fRvfUR"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-2"
  >
    <span>
      default disabled
    </span>
  </span>
</button>
`;

exports[`default error button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-25 MuiButtonBase-disabled-26 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 MuiButton-disabled-19 sc-bdVaJa hTCEyd"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-2"
  >
    <span>
      error
    </span>
  </span>
</button>
`;

exports[`default error color button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bdVaJa hTCEyd"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-2"
  >
    <span>
      error color
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-44"
  />
</button>
`;

exports[`default icon button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bdVaJa gjZMZu"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-2"
  >
    <span
      class="sc-bwzfXH czactv"
    >
      <svg
        aria-hidden="true"
        data-prefix="fal"
        data-icon="coffee"
        class="svg-inline--fa fa-coffee fa-w-18 "
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 576 512"
      >
        <path
          fill="currentColor"
          d="M517.9 448H26.1c-24.5 0-33.1-32-20-32h531.8c13.1 0 4.5 32-20 32zM576 159.1c.5 53.4-42.7 96.9-96 96.9h-32v32c0 53-43 96-96 96H160c-53 0-96-43-96-96V76c0-6.6 5.4-12 12-12h402.8c52.8 0 96.7 42.2 97.2 95.1zM416 96H96v192c0 35.3 28.7 64 64 64h192c35.3 0 64-28.7 64-64V96zm128 64c0-35.3-28.7-64-64-64h-32v128h32c35.3 0 64-28.7 64-64z"
        />
      </svg>
    </span>
    <span>
      default with icon
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-58"
  />
</button>
`;

exports[`default loading button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-25 MuiButtonBase-disabled-26 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 MuiButton-disabled-19 sc-bdVaJa lkMVbx"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-2"
  >
    <div
      class="MuiCircularProgress-root-35 MuiCircularProgress-colorPrimary-38 MuiCircularProgress-indeterminate-37"
      style="margin-right: 10px; color: rgba(191, 191, 191, 0.7);"
      role="progressbar"
    >
      <svg
        class="MuiCircularProgress-svg-40"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle-41 MuiCircularProgress-circleIndeterminate-43"
          cx="44"
          cy="44"
          r="20.2"
          fill="none"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <span>
      loading
    </span>
  </span>
</button>
`;

exports[`default secondary button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bdVaJa ffUUeT"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-2"
  >
    <span>
      secondary
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-28"
  />
</button>
`;

exports[`default success button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-25 MuiButtonBase-disabled-26 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 MuiButton-disabled-19 sc-bdVaJa ffLXpW"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-2"
  >
    <span>
      success
    </span>
  </span>
</button>
`;

exports[`default success color button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bdVaJa ffLXpW"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-2"
  >
    <span>
      success color
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-51"
  />
</button>
`;

exports[`default warning button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-bdVaJa jXMqsE"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-2"
  >
    <span />
  </span>
  <span
    class="MuiTouchRipple-root-28"
  />
</button>
`;

exports[`link default 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-153 MuiButton-root-129 MuiButton-text-131 MuiButton-flat-134 sc-bdVaJa jMXsoX"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-130"
  >
    <span>
      link
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-156"
  />
</button>
`;

exports[`link disabled button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-153 MuiButtonBase-disabled-154 MuiButton-root-129 MuiButton-text-131 MuiButton-flat-134 MuiButton-disabled-147 sc-bdVaJa KIcjQ"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-130"
  >
    <span>
      default disabled
    </span>
  </span>
</button>
`;

exports[`link icon button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-153 MuiButton-root-129 MuiButton-text-131 MuiButton-flat-134 sc-bdVaJa jMXsoX"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-130"
  >
    <span
      class="sc-bwzfXH czactv"
    >
      <svg
        aria-hidden="true"
        data-prefix="fal"
        data-icon="coffee"
        class="svg-inline--fa fa-coffee fa-w-18 "
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 576 512"
      >
        <path
          fill="currentColor"
          d="M517.9 448H26.1c-24.5 0-33.1-32-20-32h531.8c13.1 0 4.5 32-20 32zM576 159.1c.5 53.4-42.7 96.9-96 96.9h-32v32c0 53-43 96-96 96H160c-53 0-96-43-96-96V76c0-6.6 5.4-12 12-12h402.8c52.8 0 96.7 42.2 97.2 95.1zM416 96H96v192c0 35.3 28.7 64 64 64h192c35.3 0 64-28.7 64-64V96zm128 64c0-35.3-28.7-64-64-64h-32v128h32c35.3 0 64-28.7 64-64z"
        />
      </svg>
    </span>
    <span>
      default with icon
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-172"
  />
</button>
`;

exports[`link loading button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-153 MuiButtonBase-disabled-154 MuiButton-root-129 MuiButton-text-131 MuiButton-flat-134 MuiButton-disabled-147 sc-bdVaJa jFnHCu"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-130"
  >
    <div
      class="MuiCircularProgress-root-163 MuiCircularProgress-colorPrimary-166 MuiCircularProgress-indeterminate-165"
      style="margin-right: 10px; color: rgba(191, 191, 191, 0.7);"
      role="progressbar"
    >
      <svg
        class="MuiCircularProgress-svg-168"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle-169 MuiCircularProgress-circleIndeterminate-171"
          cx="44"
          cy="44"
          r="20.2"
          fill="none"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <span>
      loading
    </span>
  </span>
</button>
`;

exports[`outline default 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-89 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 sc-bdVaJa gyItIq"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-66"
  >
    <span>
      outline
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-92"
  />
</button>
`;

exports[`outline disabled button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-89 MuiButtonBase-disabled-90 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 MuiButton-disabled-83 sc-bdVaJa gyItIq"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-66"
  >
    <span>
      default disabled
    </span>
  </span>
</button>
`;

exports[`outline error button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-89 MuiButtonBase-disabled-90 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 MuiButton-disabled-83 sc-bdVaJa gyItIq"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-66"
  >
    <span>
      error
    </span>
  </span>
</button>
`;

exports[`outline error color button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-89 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 sc-bdVaJa igAhgd"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-66"
  >
    <span>
      error color
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-108"
  />
</button>
`;

exports[`outline icon button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-89 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 sc-bdVaJa gyItIq"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-66"
  >
    <span
      class="sc-bwzfXH czactv"
    >
      <svg
        aria-hidden="true"
        data-prefix="fal"
        data-icon="coffee"
        class="svg-inline--fa fa-coffee fa-w-18 "
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 576 512"
      >
        <path
          fill="currentColor"
          d="M517.9 448H26.1c-24.5 0-33.1-32-20-32h531.8c13.1 0 4.5 32-20 32zM576 159.1c.5 53.4-42.7 96.9-96 96.9h-32v32c0 53-43 96-96 96H160c-53 0-96-43-96-96V76c0-6.6 5.4-12 12-12h402.8c52.8 0 96.7 42.2 97.2 95.1zM416 96H96v192c0 35.3 28.7 64 64 64h192c35.3 0 64-28.7 64-64V96zm128 64c0-35.3-28.7-64-64-64h-32v128h32c35.3 0 64-28.7 64-64z"
        />
      </svg>
    </span>
    <span>
      default with icon
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-122"
  />
</button>
`;

exports[`outline loading button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-89 MuiButtonBase-disabled-90 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 MuiButton-disabled-83 sc-bdVaJa gyItIq"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-66"
  >
    <div
      class="MuiCircularProgress-root-99 MuiCircularProgress-colorPrimary-102 MuiCircularProgress-indeterminate-101"
      style="margin-right: 10px; color: rgb(18, 85, 102);"
      role="progressbar"
    >
      <svg
        class="MuiCircularProgress-svg-104"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle-105 MuiCircularProgress-circleIndeterminate-107"
          cx="44"
          cy="44"
          r="20.2"
          fill="none"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <span>
      loading
    </span>
  </span>
</button>
`;

exports[`outline secondary button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-89 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 sc-bdVaJa gMHwuN"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-66"
  >
    <span>
      secondary
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-92"
  />
</button>
`;

exports[`outline success button 1`] = `
<button
  tabindex="-1"
  class="MuiButtonBase-root-89 MuiButtonBase-disabled-90 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 MuiButton-disabled-83 sc-bdVaJa gyItIq"
  type="button"
  options="[object Object]"
  disabled=""
>
  <span
    class="MuiButton-label-66"
  >
    <span>
      success
    </span>
  </span>
</button>
`;

exports[`outline success color button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-89 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 sc-bdVaJa fSsuLI"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-66"
  >
    <span>
      success color
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-115"
  />
</button>
`;

exports[`outline warning button 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-89 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70 sc-bdVaJa cIBlmP"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-66"
  >
    <span />
  </span>
  <span
    class="MuiTouchRipple-root-92"
  />
</button>
`;

exports[`renders add 1`] = `
<button
  tabindex="0"
  class="MuiButtonBase-root-203 MuiButton-root-179 MuiButton-text-181 MuiButton-flat-184 sc-bdVaJa cbcxry"
  type="button"
  options="[object Object]"
>
  <span
    class="MuiButton-label-180"
  >
    <span>
      add
    </span>
  </span>
  <span
    class="MuiTouchRipple-root-206"
  />
</button>
`;
